Source/Diagnostics/ReducedDiags/ColliderRelevant.cpp

```diff
@@ -0,0 +1,555 @@
+                    const amrex::Real uy = p.rdata(PIdx::uy);
+                    const amrex::Real uz = p.rdata(PIdx::uz);
+                    const amrex::Real thetax = std::atan2(ux, uz);
+                    const amrex::Real thetay = std::atan2(uy, uz);
+                    const amrex::Real tmp1 = (x - x_ave)*(x - x_ave)*w;
+                    const amrex::Real tmp2 = (y - y_ave)*(y - y_ave)*w;
+                    const amrex::Real tmp3 = (thetax - thetax_ave)*(thetax - thetax_ave)*w;
+                    const amrex::Real tmp4 = (thetay - thetay_ave)*(thetay - thetay_ave)*w;
+                    return {tmp1, tmp2, tmp3, tmp4};
+                },
+                reduce_ops_std);
+
+            x_std = amrex::get<0>(r_std);
+            y_std = amrex::get<1>(r_std);
+            thetax_std = amrex::get<2>(r_std);
+            thetay_std = amrex::get<3>(r_std);
+
+            amrex::ParallelDescriptor::ReduceRealSum({x_std, y_std, thetax_std, thetay_std});
+
+            x_std = std::sqrt(x_std / w_tot);
+            y_std = std::sqrt(y_std / w_tot);
+            thetax_std = std::sqrt(thetax_std / w_tot);
+            thetay_std = std::sqrt(thetay_std / w_tot);
+        }
+
+        m_data[get_idx("x_ave_"+species_names[i_s])] = x_ave;
+        m_data[get_idx("x_std_"+species_names[i_s])] = x_std;
+        m_data[get_idx("y_ave_"+species_names[i_s])] = y_ave;
+        m_data[get_idx("y_std_"+species_names[i_s])] = y_std;
+        m_data[get_idx("thetax_min_"+species_names[i_s])] = thetax_min;
+        m_data[get_idx("thetax_ave_"+species_names[i_s])] = thetax_ave;
+        m_data[get_idx("thetax_max_"+species_names[i_s])] = thetax_max;
+        m_data[get_idx("thetax_std_"+species_names[i_s])] = thetax_std;
+        m_data[get_idx("thetay_min_"+species_names[i_s])] = thetay_min;
+        m_data[get_idx("thetay_ave_"+species_names[i_s])] = thetay_ave;
+        m_data[get_idx("thetay_max_"+species_names[i_s])] = thetay_max;
+        m_data[get_idx("thetay_std_"+species_names[i_s])] = thetay_std;
+#endif
+
+#if (defined WARPX_QED)
+        // get mass
+        amrex::ParticleReal m = myspc.getMass();
+        const bool is_photon = myspc.AmIA<PhysicalSpecies::photon>();
+        if (is_photon) {
+            m = PhysConst::m_e;
+        }
+
+        // compute chimin, chiave and chimax
+        amrex::Real chimin_f = 0.0_rt;
+        amrex::Real chimax_f = 0.0_rt;
+        amrex::Real chiave_f = 0.0_rt;
+
+        if (myspc.DoQED())
+        {
+            // define variables in preparation for field gatheeduce_data.value()ring
+            const int n_rz_azimuthal_modes = WarpX::n_rz_azimuthal_modes;
+            const int nox = WarpX::nox;
+            const bool galerkin_interpolation = WarpX::galerkin_interpolation;
+            const amrex::IntVect ngEB = warpx.getngEB();
+
+            // TO DO: loop over refinement levels: for (int lev = 0; lev <= level_number; ++lev)
+            int lev = 0;
```

Comment:
Shouldn't this be a `const int` for the moment?

Reply:
Thanks, done in [f4800d8](https://github.com/ECP-WarpX/WarpX/pull/4024/commits/f4800d862293e09777d30ebc1f675be886223ee5).